### PR TITLE
fix(nav2_costmap_2d): make obstacle layer not current on enabled toggle

### PR DESCRIPTION
**Pull request by bektaskemal**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | https://github.com/ros-navigation/navigation2/issues/4465 |
| Primary OS tested on | Ubuntu|
| Robotic platform tested on | Turtlebot Sim|
| Does this PR contain AI generated software? | No |

---

## Description of contribution in a few bullet points
- Made obstacle layer consistent with other layers and make current false after enabled is toggled. So isCurrent function of costmap will return false and require update.

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in docs.nav2.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_costmap_2d/plugins/obstacle_layer.cpp [hunk 1]
```diff
@@ -307,8 +307,11 @@ ObstacleLayer::dynamicParametersCallback(
         max_obstacle_height_ = parameter.as_double();
       }
     } else if (param_type == ParameterType::PARAMETER_BOOL) {
-      if (param_name == name_ + "." + "enabled") {
+      if (param_name == name_ + "." + "enabled" && enabled_ != parameter.as_bool()) {
```
Comment (SteveMacenski) [inline nav2_costmap_2d/plugins/obstacle_layer.cpp:310]: Remove, we need to do all parameter updates, this would not allow you to enable it.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Did you test this to get the behavior we discussed?

**Review by SteveMacenski — APPROVED** [review]

**Comment by bektaskemal** [comment]
> Did you test this to get the behavior we discussed?

Yes, it works as we discussed.

**Comment by SteveMacenski** [comment]
Thanks for the discussion / contribution!

**Pull request merged**